### ISSUE TITLE
Regression saveWorkbook (due to setColWidths?)

### ISSUE THREAD

**Issue by zeehio** [ycphs/openxlsx#92]
**Describe the bug**

Your dev version of openxlsx throws this error:

```
Error in .self$groupColumns(i) : object 'wb' not found
```

Causing tests in my condformat package to fail as you reported at: https://github.com/zeehio/condformat/issues/26

**To Reproduce**

I've been able to write some code that only uses `openxlsx` and reproduces the error. Run this:

```r
filename <- tempfile(fileext = ".xlsx")
x <- data.frame(a=1,b=2)
sheet_name <- "x1"

# Create a workbook and write a table:
workbook <- openxlsx::createWorkbook(creator = "")
openxlsx::addWorksheet(workbook, sheetName = "x1")
openxlsx::writeData(workbook, "x1", x, rowNames = FALSE, colNames = TRUE)
# Comment the setColWidths line below to avoid the error:
openxlsx::setColWidths(workbook, sheet = "x1", cols = seq_len(ncol(x)), widths = "auto")
openxlsx::saveWorkbook(
  workbook,
  file = filename,
  overwrite = TRUE
)

# Load the same workbook and write another table:
workbook <- openxlsx::loadWorkbook(filename)
openxlsx::addWorksheet(workbook, sheetName = "x2")
openxlsx::writeData(workbook, "x2", x, rowNames = FALSE, colNames = TRUE)
openxlsx::setColWidths(workbook, sheet = "x2", cols = seq_len(ncol(x)), widths = "auto")
openxlsx::saveWorkbook(workbook, file = filename, overwrite = TRUE)
```

There are two calls to `setColWidths` in the code above. The error does not happen if the first call to setColWidths is removed.

**Expected behavior**
An excel file is created with two sheets is found on filename

**Environment**

- Ubuntu
- R-4.0.2
- Tested on current master branch.

Thanks

**Comment by ycphs** [comment]
Hi,

I just tested your code example and there is no error popping up.

**Comment by zeehio** [comment]
Make sure you don't have any variable named `wb` on your environment. Please use a fresh R session.

I did reproduce the error on RStudio Cloud https://rstudio.cloud as well (it's free for small things, you just need to register)

![imatge](https://user-images.githubusercontent.com/75441/90398759-36f78d00-e09a-11ea-8608-9db6e26fd2d1.png)

**Comment by zeehio** [comment]
If at the beginning of the script I gave above I type: `wb <- "potato"`, the error changes to:

    Error in wb$colOutlineLevels : $ operator is invalid for atomic vectors

Meaning that the value of `wb` is not defined in that scope and it is taken instead from my `.GlobalEnv`.

This may be the reason, but I don't have time to dig further into it:

https://github.com/ycphs/openxlsx/blob/8d4d50dce3909b3dfa6c320ebe25be9833039531/R/WorkbookClass.R#L2146-L2147

Why does the second line have `wb$` and not the first one? (`wb$` appears there several times)

Thanks for your time

**Comment by ycphs** [comment]
Thank you for the hint.

The new version should fix the issue. Could you please check again?

**Comment by zeehio** [comment]
I tested against current master and it did not work.

If you meant that #91 was fixing the issue, I can't understand how it is related, since `wb` is still not defined

**Comment by zeehio** [comment]
Fixed by 343b42e52a3f4e4c210c5dcf2a13764e1b29c270.
